code/game/objects/effects/decals/cleanable.dm

```diff
@@ -15,7 +18,7 @@
 	if(loc && isturf(loc))
 		for(var/obj/effect/decal/cleanable/C in loc)
 			if(C != src && C.type == type && !QDELETED(C))
-				if (replace_decal(C))
+				if (replace_decal(C) && !bounds_dist(src, C))

```

Comment:
if bound_dists is 0 or less then the decals are overlapping or touching so this will not place a decal if its touching/overlapping the one on the same turf